src/toxcore/dht/server/mod.rs

```diff
@@ -569,28 +593,28 @@ impl Server {
         }
     }
 
-    /// Send UDP packet to specified address. The packet will be dropped if IPv6
-    /// mode is disabled but the address is IPv6 address.
-    fn send_to(&self, addr: SocketAddr, packet: DhtPacket) -> IoFuture<()> {
-        if self.is_ipv6_mode {
-            match addr.ip() {
-                IpAddr::V4(ip) => {
-                    let ip_v6 = ip.to_ipv6_mapped();
-                    let addr = SocketAddr::new(IpAddr::V6(ip_v6), addr.port());
-                    send_to(&self.tx, (packet, addr))
-                },
-                IpAddr::V6(_ip) => {
-                    send_to(&self.tx, (packet, addr))
-                },
-            }
-        } else {
-            if addr.is_ipv4() {
-                send_to(&self.tx, (packet, addr))
-            } else {
-                debug!("Attempting to send packet to IPv6 address wile DHT node is running in IPv4 mode");
-                Box::new(future::ok(()))
-            }
+    /// Send UDP packet node. If the node has both IPv4 and IPv6 addresses,
+    /// then it sends packet to both addresses.
+    fn send_to_node(&self, node: &DhtNode, packet: DhtPacket) -> IoFuture<()> {
+        let mut futures = Vec::new();
+
+        let addrs = node.get_all_sockets(self.is_ipv6_enabled);
+
+        addrs.into_iter()
+            .for_each(|addr| {
+                futures.push(send_to(&self.tx, (packet.clone(), addr)));
+            });
+
+        if futures.is_empty() {
```

Comment:
Now we can move this warning to `get_all_sockets` so that any usage of this method will make it.